tests/unit/conftest.py

```diff
@@ -27,7 +35,26 @@ class CachingAlleleTranslator(AlleleTranslator):
     def translate_from(self, var, fmt=None, **kwargs):
         return super().translate_from(var, fmt=fmt, **kwargs)
 
-    def threaded_translate_from(self, generator, num_threads=2):
+
+def caching_allele_translator_factory():
+    """Return a CachingAlleleTranslator instance with local seqrepo"""
+    dp = SeqRepoDataProxy(SeqRepo(_seqrepo_dir()))
+    assert dp is not None
+    translator = CachingAlleleTranslator(dp)
+    return translator
+
+
+class ThreadedTranslator:
+    """A class to run the translation in a threaded fashion."""
+    thread_resources = {}
+
+    def thread_initializer(self):
+        """Initialize resources for use in threads."""
+        thread = current_thread()
+        translator = caching_allele_translator_factory()
+        self.thread_resources[thread.name] = {'translator': translator}
```

Comment:
It makes sense to me that initializing multiple objects per thread to enable us to do work separately on threads, but still having trouble understanding if it conflicts with what seqrepo people said [here](https://github.com/biocommons/biocommons.seqrepo/issues/112) (the first few notes and the last two are most relevant), particularly that we didn't reach the OS file limit error that I experienced. Is this because we have a limited amount of threads?

Reply:
> multiple objects per thread 
one translator per thread seems to work, one translator shared across threads seems to fail.

> particularly that we didn't reach the OS file limit error that I experienced. Is this because we have a limited amount of threads?
What was the scenario that create OS file limit error?

From the referenced thread:
> This causes issues in applications that use SeqRepo and may attempt to fetch a sequence from the same bgz file from two threads in the same process, via the same in-process SeqRepo object. 

The key phrase here is:
> from two threads in the same process, via the `same in-process SeqRepo object`. 

We in effect have a `different SeqRepo object` per thread.

```
def caching_allele_translator_factory(normalize: bool = False, seqrepo_dir: str = _seqrepo_dir()):
    """Return a CachingAlleleTranslator instance with local seqrepo"""
    dp = SeqRepoDataProxy(SeqRepo(seqrepo_dir))
    assert dp is not None, "SeqRepoDataProxy is None"
    translator = CachingAlleleTranslator(dp)
    translator.normalize = normalize
    return translator

. . . 

    def _thread_initializer(self):
        """Initialize resources for use in threads."""
        thread = current_thread()
        translator = caching_allele_translator_factory(normalize=self.normalize)
        self._thread_resources[thread.name] = {'translator': translator}


```

Reply:
discussed offline